src/state/preferences/high-quality-images.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+
+import * as persisted from '#/state/persisted'
+
+type StateContext = persisted.Schema['highQualityImages']
+type SetContext = (v: persisted.Schema['highQualityImages']) => void
+
+const stateContext = React.createContext<StateContext>(
+  persisted.defaults.highQualityImages,
+)
+const setContext = React.createContext<SetContext>(
+  (_: persisted.Schema['highQualityImages']) => {},
+)
+
+export function Provider({children}: React.PropsWithChildren<{}>) {
+  const [state, setState] = React.useState(persisted.get('highQualityImages'))
+
+  const setStateWrapped = React.useCallback(
+    (highQualityImages: persisted.Schema['highQualityImages']) => {
+      setState(highQualityImages)
+      persisted.write('highQualityImages', highQualityImages)
+    },
+    [setState],
+  )
+
+  React.useEffect(() => {
+    return persisted.onUpdate('highQualityImages', nextHighQualityImages => {
+      setState(nextHighQualityImages)
+    })
+  }, [setStateWrapped])
+
+  return (
+    <stateContext.Provider value={state}>
+      <setContext.Provider value={setStateWrapped}>
+        {children}
+      </setContext.Provider>
+    </stateContext.Provider>
+  )
+}
+
+export function useHighQualityImages() {
+  return React.useContext(stateContext)
+}
+
+export function useSetHighQualityImages() {
+  return React.useContext(setContext)
+}
+
+// This is a little weird to have here imo but it works I guess
```

Comment:
agree its probably fine to have this here, gonna need a different place as part of synced preferences but whatever